src/main/java/de/adesso/kicker/user/UserController.java

```diff
@@ -21,7 +22,13 @@ public UserController(UserService userService) {
     @GetMapping("/users/{id}")
     public ModelAndView getUser(@PathVariable String id) {
         ModelAndView modelAndView = new ModelAndView();
-        modelAndView.addObject("user", userService.getUserById(id));
+        try {
+            modelAndView.addObject("user", userService.getUserById(id));
+        }
+        catch (UserNotFoundException u) {
```

Comment:
This still gives the user a `200 OK` status code. You can add `@ResponseStatus(code = HttpStatus.NOT_FOUND)` to the exception itself and not catch it. Spring will automatically return the 404 page and give the proper error code.

Reply:
I will fix it in Issue#46.